packages/foam-vscode/src/core/services/markdown-provider.ts

```diff
@@ -19,13 +19,13 @@ export class MarkdownResourceProvider implements ResourceProvider {
 
   constructor(
     private readonly matcher: IMatcher,
+    private readonly dataStore: IDataStore,
```

Comment:
I made this required so that an explicit choice needs to be made on how files should be read